src/Microsoft.OData.Core/JsonLight/ODataJsonLightReader.cs

```diff
@@ -202,11 +202,8 @@ protected override Task<bool> ReadAtStartImplementationAsync()
                 this.jsonLightInputContext.CreatePropertyAndAnnotationCollector();
 
             // Position the reader on the first node depending on whether we are reading a nested payload or not.
-            ODataPayloadKind payloadKind = this.ReadingDelta
-                ? ODataPayloadKind.Delta
-                : this.ReadingResourceSet ?
-                    ODataPayloadKind.ResourceSet
-                    : ODataPayloadKind.Resource;
+            ODataPayloadKind payloadKind = this.ReadingResourceSet ?
+                this.ReadingDelta ? ODataPayloadKind.Delta : ODataPayloadKind.ResourceSet : ODataPayloadKind.Resource;

```

Comment:
I think this is less readable than the old code. Maybe a bit of nesting might help:
```suggestion
            ODataPayloadKind payloadKind = this.ReadingResourceSet ?
                this.ReadingDelta ? ODataPayloadKind.Delta : ODataPayloadKind.ResourceSet
                : ODataPayloadKind.Resource;

```